### PR TITLE
[Draft] feat(planner): Add .NET support

### PR DESCRIPTION
**Pull request by absundr**
## Feature description
Add support for .NET applications

## Solution description
With this PR, you will be able to generate docker images for dotnet applications. It includes test cases with a sample dotnet application. Please note that this is still a minimum implementation, and it supports simple dotnet applications. In future, we can add more revisions to support more complex dotnet applications with multiple build paths. 

## Tested frameworks
1. Dotnet console 
2. ASP.NET Web App
3. ASP.NET Web API

**Not tested for older .NET framework versions**

Please review, feedback is appreciated. 

Relates to #64

### CODE REVIEW

**Comment by absundr** [comment]
> Nice work! It seems that you failed one of the tests? Besides, I need @pan93412 for further review. Would you like to join our Discord channel for more discussions? https://discord.gg/DrdGCvXEyY

Yeah, it looks like I forgot to lint and format before committing. It's fixed now and yes, I'd love to join the discord!

**Comment by pan93412** [comment]
```
You have commit messages with warnings

⧗   input: fix(lint): lint and format
⚠   subject must be sentence-case, start-case [subject-case]

⚠   found 0 problems, 1 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
```

Sentence case: `fix(lint): Lint and Format`.
`git rebase -i HEAD^^` to update the commit message.

File: internal/dotnet/identify.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+func NewIdentifier() plan.Identifier {
+	return &identify{}
+}
+
+func (i *identify) PlanType() types.PlanType {
+	return types.PlanTypeDotnet
+}
+
+func (i *identify) Match(fs afero.Fs) bool {
+	return utils.HasFile(fs, "Program.cs", "Startup.cs")
```
Comment (pan93412) [inline internal/dotnet/identify.go:23]: Can it support other project types than `ASP.NET`?

Comment (absundr) [reply internal/dotnet/identify.go:23]: I've tested it for ASP.NET Web Apps(Blazor & MVC), ASP.NET Web APIs, and ASP.NET Microservices. I haven't tested with .NET Framework apps and it won't work atm.  I have sample projects that I created for them, do you want me to add it to the `tests` folder?

Keep in mind that I did not take into consideration more complex applications with multiple modules and when the `.csproj` isn't present in the root. We need a way to parse the `.sln` file, read the modules in the project, and add it to the docker template. I'm open to any ideas if you know of a good way to do this.

**Review by pan93412 — APPROVED** [review]

**Review by yuaanlin — APPROVED** [review]
Nice work!

**Pull request merged**